library/src/main/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapter.java

```diff
@@ -600,6 +600,28 @@ public int getFooterPositionInAdapter(Section section) {
         return getSectionPosition(section) + section.getSectionItemsTotal() - 1;
     }
 
+    /**
+     * Returns the index of the first occurrence of the specified section in this adapter, or -1 if
+     * this adapter does not contain the section. Note that, the visibility of section is being
+     * ignored.
+     *
+     * @param section section to search for
+     * @return the index of the first occurrence of the specified section in this adapter, or -1 if
+     *         this adapter does not contain the section
+     */
+    public int indexOf(Section section) {
+        int index = 0;
+
+        for (Map.Entry<String, Section> entry : this.sections.entrySet()) {
+            if (entry.getValue() == section) {
+                return index;
+            }
+            index++;
+        }
+
+        return -1;
```

Comment:
could you please throw an IllegalArgumentException, update javadoc and add a unit test for this scenario?

Reply:
Hi @luizgrp 

Do you want to function to have similar behavior of ArrayList's `indexOf` ? (It returns -1 for not found case)

https://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html#indexOf(java.lang.Object)

However, if you prefer to have `IllegalArgumentException`, I'm fine with that too, although I prefer it to have similar behavior as ArrayList's `indexOf`. (Due to principle of least astonishment)

Reply:
sure, let's keep it returning -1 👍